src/operator/nn/mkldnn/mkldnn_batch_norm-inl.h

```diff
@@ -234,20 +234,15 @@ void MKLDNNBatchNormForward(const OpContext &ctx, const BatchNormParam &param,
     DType* weight_ptr = gamma.data().dptr<DType>();
     DType* bias_ptr = beta.data().dptr<DType>();
     if (!param.fix_gamma) {
-#pragma omp parallel for
-      for (int i = 0; i < channels_; i++) {
-        weight_buf[i] = weight_ptr[i];
-        weight_buf[channels_ + i] = bias_ptr[i];  // bias
-      }
+      memcpy(weight_buf, weight_ptr, sizeof(weight_buf[0]) * channels_);
+      memcpy(&weight_buf[channels_], bias_ptr, sizeof(weight_buf[0]) * channels_);
```

Comment:
nice optimization above; why is the below OMP calls causing overhead.

Reply:
are you referring to all of the OMP directives? The number of channels is in the order of 100. Parallelization overhead is usually larger than the actual computation.

Reply:
got it, thanks! we noticed the same performance issue for smaller networks too (eg: mnist) . Lower OMP_NUM_THREADS (eg: 4 -vs- 36) was giving better performance.